doc/developer/design/20230214_transform_using.md

```diff
@@ -0,0 +1,274 @@
+# Overview
+
+*Allow users to transform a source before persisting it using SQL*
+
+# Goals
+
+- Make append-only sources viable by retaining a bounded number of updates
+- Thin-down sources by filtering before consuming storage
+
+This design doc proposes supporting arbitrary data transformations during ingestion made available
+to users by extending the syntax of the `CREATE SOURCE` statement. The goal is to allow expressing
+any transformation that can be written as a SQL query with one candidate syntax being a `TRANSFORM
+USING <query>` fragment that can be optionally specified at the end of a `CREATE SOURCE` statement.
+
+**Example**
+
+```sql
+CREATE SOURCE foobar FROM POSTGRES ...
```

Comment:
The examples are somewhat misleading because the "source" `foobar` doesn't carry any data for postgres sources. Here we would have to have a `TRANSFORM USING` per table, it seems?